microprofile/openapi/src/main/java/io/helidon/microprofile/openapi/MPOpenAPIBuilder.java

```diff
@@ -172,14 +172,11 @@ MPOpenAPIBuilder config(Config mpConfig) {
     }
 
     private static Optional<? extends Application> instantiate(Class<? extends Application> appClass) {
-        try {
-            return Optional.of(appClass.getConstructor().newInstance());
-        } catch (InstantiationException | IllegalAccessException | InvocationTargetException e) {
-            throw new RuntimeException(e);
-        } catch (NoSuchMethodException e) {
-            // Wrapper app does not have a no-args constructor so we canont instantiate it.
-            return Optional.empty();
-        }
+        Unmanaged<? extends Application> unmanagedApp = new Unmanaged<>(appClass);
+        Unmanaged.UnmanagedInstance<? extends Application> unmanagedInstance = unmanagedApp.newInstance();
+        Application app = unmanagedInstance.produce().inject().postConstruct().get();
+        unmanagedInstance.preDestroy().dispose();
+        return Optional.of(app);
```

Comment:
Since disposal will do nothing, this is probably OK, but it looks weird.  I wonder if you can move this `Unmanaged` stuff elsewhere, like where you call something like `getClasses()` on the `Application` instance, and do the disposal afterwards.

Reply:
It sounds as if your concern is that the use of the instance (the `app`) outlives the `unmanagedInstance` that gave rise to it. 

I see how that does look a bit odd. But the app instance also outlives the code around line 123 and in fact is used not only in the same `MPOpenAPIBuilder` but also in a separate class `OpenApiCdiExtension.` 

I'll take a look at refactoring things to address that admittedly odd-looking code without altering too much the way the consuming code operates. Thanks for the catch.

Reply:
Yeah, the main problem is that dependent objects that may get injected into the `Application` instance (and of course it's user-supplied so you don't have any control over it) may get flushed/released/disposed when that `dispose()` call is made.  So you don't want them to go away obviously while the `Application` instance is in use somewhere.

Reply:
The latest push rearranges the code so it uses any unmanaged instance as much as needed, and only then disposes of it.